src/Illuminate/Database/Migrations/Migrator.php

```diff
@@ -557,7 +568,9 @@ public function getMigrationFiles($paths)
     public function requireFiles(array $files)
     {
         foreach ($files as $file) {
-            $this->files->requireOnce($file);
```

Comment:
`static::$requiredMigrations[$file] `??=` $this->files->requireOnce($file);`

Try this instead of IF